lib/inc/internal/facts/linux/virtualization_resolver.hpp

```diff
@@ -7,20 +7,36 @@
 #include "../resolvers/virtualization_resolver.hpp"
 #include <string>
 
+using namespace std;
+
 namespace facter { namespace facts { namespace linux {
 
     /**
      * Responsible for resolving virtualization facts.
      */
     struct virtualization_resolver : resolvers::virtualization_resolver
     {
-     protected:
+     public:
         /**
          * Gets the name of the hypervisor.
          * @param facts The fact collection that is resolving facts.
          * @return Returns the name of the hypervisor or empty string if no hypervisor.
          */
-        virtual std::string get_hypervisor(collection& facts) override;
+        std::string get_hypervisor(collection& facts) override;
+        /**
+         * Gets the name of the cloud provider.
+         * @param facts The fact collection that is resolving facts.
+         * @return Returns the name of the cloud provider or empty string if no hypervisor.
+         */
+        std::string get_cloud_provider(collection& facts) override;
+
+        /**
+         * Gets whether the machine is running in Azure.
+         * @param facts The fact collection that is resolving facts.
+         * @param leases_file The location of where the leases file exists.
+         * @return Returns "azure" if running on azure, otherwise an empty string.
+         */
+        static std::string get_azure(collection& facts, string const& leases_file = "/var/lib/dhcp/dhclient.eth0.leases");
```

Comment:
There is a [Powershell script](https://gallery.technet.microsoft.com/scriptcenter/Detect-Windows-Azure-aed06d51) that exists already; I believe it just needs to be converted into C++.